lib/location.js

```diff
@@ -106,13 +101,12 @@ var LocationState = State.extend(function (self, name, opts) {
     self.characters_per_page = opts.characters_per_page;
     self.next_text = opts.next_text;
     self.previous_text = opts.previous_text;
-    self.store_fields = opts.store_fields;
-    self.gmapsURL = 'http://maps.googleapis.com/maps/api/geocode/json';
     self.namespace = opts.namespace;
+    self.map_provider = opts.map_provider || new providers.GoogleMaps();
 
     self.init = function() {
         _.defaults(self.metadata, {locationstate:'initial'});
-        self.http = new JsonApi(self.im);
+        self.map_provider.init(self.im);
```

Comment:
Wondering if a provider's `.init` will ever return a promise, in which case we would need to handle that here. Not too important though.